lib/client.js

```diff
@@ -385,17 +351,66 @@ class Client extends EventEmitter {
         callback(err)
       }
     })
+    ret[kNeedDrain] = false
+    ret.write = function (chunk, encoding, callback) {
+      // TODO: encoding, callback
+      // TODO: ERR_STREAM_WRITE_AFTER_END, ERR_STREAM_DESTROYED
+      const needDrain = !req.push(chunk)
+      if (needDrain) {
+        ret[kNeedDrain] = true
+      }
+      return needDrain
+    }
+    ret.end = function (chunk, encoding, callback) {
+      // TODO: encoding, callback
+      // TODO: ERR_STREAM_ALREADY_FINISHED, ERR_STREAM_DESTROYED
```

Comment:
Right now this is only intendend to be used with `stream.pipeline` where these TODO's are not entirely relevant.